openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -108,11 +111,26 @@ func CreateCinderVolume(vs *gophercloud.ServiceClient, options controller.Volume
 // become available.  The connection information cannot be retrieved from cinder
 // until the volume is available.
 func WaitForAvailableCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
-	// TODO: Implement proper polling instead of brain-dead timers
 	c := make(chan error)
-	go time.AfterFunc(5*time.Second, func() {
-		c <- nil
-	})
+	go func() error {
```

Comment:
could we use util function `k8s.io/apimachinery/pkg/util/wait.poll()` instead? @chestack @aglitke

Reply:
@wenjun93 +1

Reply:
+1

Reply:
thanks for all your comments. Done in next patch. sorry for fixing it late.